### PR TITLE
Check exact hierarchy of Exception classes.

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This spec checks the hierarchy of exception classes.  It uses the precise sub.superclass, not the less precise be_ancestor_of.

This PR, if merged, would allow removal of be_ancestor_of throughout the exception specs.

### CODE REVIEW

File: core/exception/hierarchy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                NameError => {
+                    NoMethodError => nil,
+                },
+                RangeError => {
+                    FloatDomainError => nil,
+                },
+                RegexpError => nil,
+                RuntimeError => nil,
+                SystemCallError => nil,
+            ThreadError => nil,
```
Comment (eregon) [inline core/exception/hierarchy_spec.rb]: The indentation went wrong here and below

File: core/exception/hierarchy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+require_relative '../../spec_helper'
+
+describe "Exception" do
+  it "has the right class hierarchy" do
+    def save_pairs(parent, hash, pairs)
```
Comment (eregon) [inline core/exception/hierarchy_spec.rb]: Could you make this a lambda and move it below the `hierarchy = {`?

Comment (eregon) [reply core/exception/hierarchy_spec.rb]: Tracking the `pairs` explicitly seems extra complexity, I would just use `should` inside this lambda.

File: core/exception/hierarchy_spec.rb (hunk 1 above)
Comment (BurdetteLamar) [reply core/exception/hierarchy_spec.rb]: Oops.  Adjusted the setting in RubyMine.

File: core/exception/hierarchy_spec.rb (hunk 2 above)
Comment (BurdetteLamar) [reply core/exception/hierarchy_spec.rb]: Done (I think).  Learned about lambdas.  (Learning is my "pay.")

**Pull request merged**